### PR TITLE
fix: support for multiple subarrays/cmc's

### PR DESCRIPTION
**Pull request by mmphego**
- Updated `sensor_poll.py` by means of brute force, sensor_values stored as ```<hostname>.<subarray-name>.sensor_values.json``` 
- Reworked the way assertions are handled. `raise AssertionError()` is ideal when running `python -O` if one wants to optimize the bytecode - which is unnecessary for this script.

### CODE REVIEW

**Comment by kmadisa** [comment]
@mmphego Please add a description.

**Comment by mmphego** [comment]
> @mmphego Please add a description.

You might as well review, the whole script.
But be warned, The code is brutal - I needed to get something working before I started refactoring.

File: src/sensor_poll.py [hunk 1]
```diff
@@ -380,12 +370,12 @@ def new_mapping(self, _host):
...
             self.logger.debug("Sorting ordered sensor dict by %ss!!!" % _host)
             ordered_sensor_dict = self.get_sensor_dict
-            if not isinstance(ordered_sensor_dict, OrderedDict):
-                raise AssertionError()
+            assert isinstance(ordered_sensor_dict, OrderedDict)
         except Exception:
             self.logger.error("Failed to retrieve sensor dict", exc_info=True)
         else:
             mapping = []
+            # import IPython; globals().update(locals()); IPython.embed(header='Python Debugger')
```
Comment (kmadisa) [inline src/sensor_poll.py]: Remove commented code.

Comment (mmphego) [reply src/sensor_poll.py]: Done!

File: src/sensor_poll.py [hunk 2]
```diff
@@ -80,6 +82,10 @@ def __init__(self, katcp_ip="10.103.254.6", katcp_port=7147):
             Port to connect to! [Defaults: 7147]
         """
         self.katcp_ip = katcp_ip
+        try:
+            self.hostname = ''.join(socket.gethostbyaddr(katcp_ip)[1])
+        except Exception:
+            self.hostname = 'Unknown'
```
Comment (kmadisa) [inline src/sensor_poll.py:88]: If `hostname == 'Unknown'` do you still want to invoke `_kcp_connect`?

Comment (mmphego) [reply src/sensor_poll.py:88]: Yes, with an unknown hostname.

**Pull request merged**